test/posts.test.js

```diff
@@ -58,3 +62,104 @@ describe('test /posts endpoint', () => {
     expect(res.body).toBe(150);
   });
 });
+
+describe('test /posts/:guid responses', () => {
+  // an array of keys.
+  const existingGuid = 'existing-guid';
+  const missingGuid = 'missing-guid';
+  const existingNeedingEncoding = 'guid needing/encoding';
+  const missingNeedingEncoding = 'missing needing/encoding';
+
+  // an object to be added for testing purposes
+  const addedPost1 = new Post(
+    'foo',
+    'foo',
+    '',
+    'foo',
+    new Date('2009-09-07T22:20:00.000Z'),
+    new Date('2009-09-07T22:23:00.000Z'),
+    'foo',
+    'foo',
+    existingGuid
+  );
+
+  // an object, expected to be returned by a correct query
+  const receivedPost1 = {
+    author: 'foo',
+    title: 'foo',
+    html: '',
+    text: 'foo',
+    published: '2009-09-07T22:20:00.000Z',
+    updated: '2009-09-07T22:23:00.000Z',
+    url: 'foo',
+    site: 'foo',
+    guid: 'existing-guid',
+  };
+
+  // an object to be added for testing purposes
+  const addedPost2 = new Post(
+    'bar',
+    'bar',
+    '',
+    'bar',
+    new Date('2010-09-07T22:20:00.000Z'),
+    new Date('2010-09-07T22:23:00.000Z'),
+    'bar',
+    'bar',
+    existingNeedingEncoding
+  );
+
+  // an object, expected to be returned by a correct query
+  const receivedPost2 = {
+    author: 'bar',
+    title: 'bar',
+    html: '',
+    text: 'bar',
+    published: '2010-09-07T22:20:00.000Z',
+    updated: '2010-09-07T22:23:00.000Z',
+    url: 'bar',
+    site: 'bar',
+    guid: 'guid needing/encoding',
+  };
+
+  // add the post to the storage
+  beforeAll(async () => {
+    addedPost1.save();
+    addedPost2.save();
+  });
+
+  // tests
+  it("pass a guid that doesn't exist", async () => {
+    const res = await request(app).get(`/posts/${missingGuid}`);
+
+    expect(res.status).toEqual(404);
+    expect(res.get('Content-type')).toContain('application/json');
+    expect(res.body instanceof Object).toBe(true);
```

Comment:
This won't work, since an Array is an Object:

```
> [] instanceof Object
true
```

Instead of adding this particular test to everything below, just do one test that makes sure that it's *not* an Array.

Reply:
fixed.